superdesk/content_types_async/service.py

```diff
@@ -0,0 +1,581 @@
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2025 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+from copy import deepcopy
+from typing import Any, Literal, cast
+
+import bson
+from bson import ObjectId
+from quart_babel import gettext as _
+
+from apps.auth import get_user_id
+from apps.desks import remove_profile_from_desks
+from apps.desks_async.desks_async_service import DesksAsyncService
+from apps.templates.content_templates import ContentTemplatesService
+import superdesk
+from superdesk.core.resources.service import AsyncCacheableService
+from superdesk.core.types.search import ProjectedFieldArg, SearchRequest
+from superdesk.default_schema import DEFAULT_EDITOR, DEFAULT_SCHEMA, DEFAULT_SCHEMA_MAP
+from superdesk.errors import SuperdeskApiError
+from superdesk.resource_fields import ID_FIELD
+from superdesk.types.content_types import ContentTypes
+from superdesk.utc import utcnow
+from superdesk.utils import format_content_type_name
+from superdesk.vocabularies_async.service import VocabulariesService
+
+
+CONTENT_TYPE_PRIVILEGE = "content_type"
+DO_NOT_SHOW_SELECTION = "do not show"
+
+# Fields that might not be in the schema but should be still available in formatter/output
+REQUIRED_FIELDS = ("language", "embargoed")
+
+# Valid editor keys
+EDITOR_ATTRIBUTES = (
+    "order",
+    "sdWidth",
+    "required",
+    "readonly",
+    "hideDate",
+    "showCrops",
+    "formatOptions",
+    "editor3",
+    "default",
+    "cleanPastedHTML",
+    "imageTitle",
+    "sourceField",
+    "section",
+    "preview",
+    "enabled",
+    "field_name",
+    "allow_toggling",
+)
+
+# cvs hardcoded in the app wich special use
+# and not supposed to be added to content profile
+HARDCODED_CVS = ("languages",)
+
+
+class ContentTypesService(AsyncCacheableService[ContentTypes]):
+    resource_name = "content_types"
+
+    async def _set_created_by(self, doc: ContentTypes) -> None:
+        doc.created_by = await get_user_id()
```

Comment:
Is `get_user_id` async yet, otherwise this would raise an error stating can't await as the function is not a coroutine